llarp/net/route.cpp

```diff
@@ -338,28 +340,31 @@ namespace llarp::net
   {
     std::vector<std::string> gateways;
 #ifdef __linux__
-
-    FILE* p = popen("ip route", "r");
-    if (p == nullptr)
-      return gateways;
-    char* line = nullptr;
-    size_t len = 0;
-    ssize_t read = 0;
-    while ((read = getline(&line, &len, p)) != -1)
-    {
-      std::string line_str(line, len);
-      std::vector<std::string> words;
-      std::istringstream instr(line_str);
-      for (std::string word; std::getline(instr, word, ' ');)
+    // return true when it's all zeros
+    const auto is_zero = [](std::string_view str) {
+      for (const auto& ch : str)
       {
-        words.emplace_back(std::move(word));
+        if (ch != '0')
+          return false;
       }
-      if (words[0] == "default" and words[1] == "via" and words[3] == "dev" and words[4] != ifname)
+      return true;
+    };
+    std::ifstream inf("/proc/net/route");
+    for (std::string line; std::getline(inf, line);)
+    {
+      const auto parts = split(line, '\t');
+      if (is_zero(parts[1]) and parts[0] != ifname)
```

Comment:
The lambda could be ditched and replaced with:
```C++
if (parts[1].find_first_not_of('0') != std::string::npos and ...
```

Reply:
Oops, should be `==`

Reply:
done

Reply:
waiting for push before approval

Reply:
pushed